### DISCUSSION TITLE
InBrowser USB Serial monitor ESP, colored weblinks addon

### DISCUSSION THREAD

**Discussion by ldijkman** [sourcelair/xterm.js#4967] [Q&A]
Hello 

i have an inbrowser serial usb monitor
it is connected to ESP8266 or ESP32 device
it sends over serial usb a mDNS scanlist of other devices in local wifi network webservers

links are clickable in serial monitor

but is it possible to give them another color, so that they are better vissible as being links

have been searching and playing with chatgpt but cannot get it working colored
 yet
greet luberth

https://ldijkman.github.io/async-esp-fs-webserver/WebSerialMonitor.html

![Screenshot from 2024-02-24 04-32-55](https://github.com/xtermjs/xterm.js/assets/45427770/b773a102-4263-41e5-b562-cb5bacaac21b)

![Screenshot from 2024-02-24 04-50-48](https://github.com/xtermjs/xterm.js/assets/45427770/5adbfc5a-492f-4502-a47e-79e4e4a67757)

**Comment by jerch** [comment]
Well since xterm.js is a terminal, the most straight forward way to achieve coloring&underline by default is to apply it to the incoming data stream already. E.g. identify URLs in your data and decorate them with terminal sequences (`\x1b[31;4m<your URL>\x1b[m` - would write them in red and single underlined).
